net/socket/recvmsg.c

```diff
@@ -88,11 +88,6 @@ ssize_t psock_recvmsg(FAR struct socket *psock, FAR struct msghdr *msg,
       return -EINVAL;
     }
 
-  if (msg->msg_iovlen != 1)
```

Comment:
why not implement multi message recv in this function?

Reply:
it's the work for the furture improvement, your contribution is welcome.

Reply:
let us hold this commit. the protocol layer does not have multi-segment data mapping implement, so I think to resolve the multi-segment reception problem in VFS, rather than requiring similar changes for each protocol.

Reply:
do you review the recent change from @yamt : https://github.com/apache/nuttx/pull/14597 https://github.com/apache/nuttx/pull/13498 https://github.com/apache/nuttx/pull/12674
it's impossible to implement the multi-segment data transfer on read(recv)/write(send) without the additional allocation.

Reply:
why? The current code has implemented the compatibility mode:
https://github.com/apache/nuttx/blob/master/fs/vfs/fs_read.c#L53

@yamt why do we need to introduce the implementation of uio? Only 1 character reading and writing are used in the code, which will consume more CPU cycles:
https://github.com/apache/nuttx/blob/master/drivers/serial/serial.c#L1009

Reply:
> why? The current code has implemented the compatibility mode: https://github.com/apache/nuttx/blob/master/fs/vfs/fs_read.c#L53
> 

but the implementation isn't a full compatibility, and can't be used in socket case at all.

Reply:
I plan to use sockcap properties to solve this problem and will submit a patch tomorrow.